code/controllers/subsystem/pois.dm

```diff
@@ -0,0 +1,186 @@
+	if(new_poi.client?.holder?.fakekey)
+		return
+
+	RegisterSignal(new_poi, COMSIG_PARENT_QDELETING, .proc/on_mob_qdel)
+
+	if(isnewplayer(new_poi))
+		lobby_points_of_interest += new_poi
+	else
+		mob_points_of_interest += new_poi
+
+	on_pois_changed(POI_MOBS)
+
+/// Remove old_poi from the mob POI list and unregister signals.
+/datum/controller/subsystem/pois/proc/del_mob_poi(mob/old_poi)
+	UnregisterSignal(old_poi, COMSIG_PARENT_QDELETING)
+
+	if(isnewplayer(old_poi))
+		lobby_points_of_interest -= old_poi
+	else
+		mob_points_of_interest -= old_poi
+
+	on_pois_changed(POI_MOBS)
+
+/// Adds new_poi to the other POI list and registers key signals if necessary.
+/datum/controller/subsystem/pois/proc/add_other_poi(atom/new_poi)
+	RegisterSignal(new_poi, COMSIG_PARENT_QDELETING, .proc/on_other_qdel)
+	other_points_of_interest += new_poi
+	on_pois_changed(POI_OTHER)
+
+/// Remove old_poi from the other POI list and unregister signals.
+/datum/controller/subsystem/pois/proc/del_other_poi(atom/old_poi)
+	UnregisterSignal(old_poi, COMSIG_PARENT_QDELETING)
+	other_points_of_interest -= old_poi
+	on_pois_changed(POI_OTHER)
+
+/// Signal handler for when mobs are created.
+/datum/controller/subsystem/pois/proc/on_glob_mob_created(datum/source, mob/created_mob)
+	SIGNAL_HANDLER
+	add_mob_poi(created_mob)
+
+/// Signal handler for when mobs who we consider POIs get deleted.
+/datum/controller/subsystem/pois/proc/on_mob_qdel(datum/deleted_mob, force)
+	SIGNAL_HANDLER
+	del_mob_poi(deleted_mob)
+
+/// Signal handler for when non-mob POIs have been deleted.
+/datum/controller/subsystem/pois/proc/on_other_qdel(datum/deleted_poi, force)
+	SIGNAL_HANDLER
+	del_other_poi(deleted_poi)
+
+/// Signal handler for when the POI element gets attached to something.
+/datum/controller/subsystem/pois/proc/on_glob_poi_created(datum/source, atom/new_poi)
+	SIGNAL_HANDLER
+
+	if(ismob(new_poi))
+		add_mob_poi(new_poi)
+		return
+
+	add_other_poi(new_poi)
+
+/// Signal handler for when the POI element gets removed from something
+/datum/controller/subsystem/pois/proc/on_glob_poi_removed(datum/source, atom/former_poi)
+	SIGNAL_HANDLER
+	if(ismob(former_poi))
+		del_mob_poi(former_poi)
+		return
+
+	del_other_poi(former_poi)
+
+/// Signal handler for when new stealthmin created. Removes them as a mob POI.
+/datum/controller/subsystem/pois/proc/on_glob_stealthmin_gained(datum/source, mob/new_stealthmin)
+	SIGNAL_HANDLER
+
+	del_mob_poi(new_stealthmin)
+
+/// Signal handler for when stealthmin ends, re-adds them as a mob POI.
+/datum/controller/subsystem/pois/proc/on_glob_stealthmin_lost(datum/source, mob/former_stealthmin)
+	SIGNAL_HANDLER
+
+	add_mob_poi(former_stealthmin)
+
+/// Send a signal to indicate that this SS's POI lists have changed. Listeners can then update.
+/datum/controller/subsystem/pois/proc/on_pois_changed(poi_type)
+	switch(poi_type)
+		if(POI_MOBS)
+			SEND_SIGNAL(src, COMSIG_MOB_POIS_CHANGED)
+			return
```

Comment:
The returns here are not necessary, we don't have falling switches.

Reply:
This proc has been removed in the latest version.